### PR TITLE
Eks basic integration tests for cloudbeat

### PR DESCRIPTION
**Pull request by gurevichdmitry**
Added support for executing cloudbeat integration tests on EKS.
Tests execution - **Production Mode** (Pytest docker is running inside EKS cluster)
1. Provision EKS cluster using aws or eksctl tools
2. Select EKS context `kubectl config use-context <your_eks_context>`
3. Deploy tests environment: `just deploy-local-tests-helm pre_merge`
4. Apply AWS certificates: `kubectl create secret generic aws-creds-secret -n kube-system --from-literal=AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID --from-literal=AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY `
5. Execute tests `just run-tests`
6. If all tests passed test-pod status will be succeeded, otherwise read logs from test-pod.
 
Tests execution - **Tests Dev Mode** (Tests execution is done on local machine)
1. Repeat step 1-4 from **Production mode**
2. Perform port forward for ES: `kubectl port-forward svc/elasticsearch-master -n kube-system 9200`
3. Execute tests `poetry run pytest -s -v -m "pre_merge" --alluredir=./allure/results/ --clean-alluredir`

If changing cloudbeat code:
- Build Cloudbeat image and upload to ECR, note that for chip M1 specify GOARCH=amd64 and for docker -> --platform linux/amd64

If changing tests code:
- Build pytest docker `just build-pytest-docker`
- Load pytest docker to ECR `just load-pytest-eks`

### CODE REVIEW

**Comment by oren-zohar** [comment]
> Build Cloudbeat image and upload to ECR, note that for chip M1 specify GOARCH=amd64 and for docker -> --platform linux/amd64

@gurevichdmitry this seems counter-intuitive, can you explain?

**Comment by gurevichdmitry** [comment]
@oren-zohar, sure. If creating default EKS cluster, aws tool uses Amazon Linux 2 (x86_64) image for cluster node which is amd64.

Current just command will build arm64 on M1 chip:
`GOOS=linux go build -v && docker build -t cloudbeat .`

In order to create amd64:
`GOOS=linux GOARCH=amd64 go build -v && docker build --platform linux/amd64 -t cloudbeat .`

File: tests/commonlib/elastic_wrapper.py [hunk 1]
```diff
@@ -53,7 +53,10 @@ def get_doc_source(data: dict, index: int = 0) -> dict:
         @param index: Document index in result dictionary
         @return: Source dictionary
         """
-        ret_value = data['hits']['hits'][index]['_source']
+        try:
+            ret_value = data['hits']['hits'][index]['_source']
+        except IndexError:
+            return {}
```
[the remark this answers is not in the document]
Comment (gurevichdmitry) [reply tests/commonlib/elastic_wrapper.py:59]: Fixed.
Added simple print function, and later we will be replaced by logger.

File: tests/commonlib/utils.py [hunk 2]
```diff
@@ -22,13 +23,15 @@ def get_ES_evaluation(elastic_client, timeout, rule_tag, exec_timestamp,
 
     while time.time() - start_time < timeout:
         try:
-            events = get_events_from_index(elastic_client, elastic_client.index, rule_tag, latest_timestamp)
+            events = get_events_from_index(
```
[the remark this answers is not in the document]
Comment (gurevichdmitry) [reply tests/commonlib/utils.py:26]: Fixed

File: tests/commonlib/utils.py [hunk 3]
```diff
@@ -22,13 +23,15 @@ def get_ES_evaluation(elastic_client, timeout, rule_tag, exec_timestamp,
...
+                elastic_client, elastic_client.index, rule_tag, latest_timestamp)
         except Exception as e:
             print(e)
             continue
 
         for event in events:
-            findings_timestamp = datetime.datetime.strptime(getattr(event, '@timestamp'), '%Y-%m-%dT%H:%M:%S.%fZ')
+            findings_timestamp = datetime.datetime.strptime(
+                getattr(event, '@timestamp'), '%Y-%m-%dT%H:%M:%S.%fZ')
```
[the remark this answers is not in the document]
Comment (gurevichdmitry) [reply tests/commonlib/utils.py:34]: Fixed

File: tests/commonlib/utils.py [hunk 4]
```diff
@@ -61,13 +64,15 @@ def get_logs_evaluation(k8s, timeout, pod_name, namespace, rule_tag, exec_timest
     start_time = time.time()
     while time.time() - start_time < timeout:
         try:
-            logs = get_logs_from_stream(k8s.get_pod_logs(pod_name=pod_name, namespace=namespace, since_seconds=2))
+            logs = get_logs_from_stream(k8s.get_pod_logs(
+                pod_name=pod_name, namespace=namespace, since_seconds=2))
```
[the remark this answers is not in the document]
Comment (gurevichdmitry) [reply tests/commonlib/utils.py:68]: fixed

File: tests/commonlib/utils.py [hunk 5]
```diff
@@ -61,13 +64,15 @@ def get_logs_evaluation(k8s, timeout, pod_name, namespace, rule_tag, exec_timest
...
         except Exception as e:
             print(e)
             continue
 
         for log in logs:
-            findings_timestamp = datetime.datetime.strptime(log.time, '%Y-%m-%dT%H:%M:%Sz')
+            findings_timestamp = datetime.datetime.strptime(
+                log.time, '%Y-%m-%dT%H:%M:%Sz')
```
[the remark this answers is not in the document]
Comment (gurevichdmitry) [reply tests/commonlib/utils.py:75]: fixed

File: tests/commonlib/utils.py [hunk 6]
```diff
@@ -125,32 +130,35 @@ def wait_for_cycle_completion(elastic_client, nodes: list) -> bool:
...
     curr_sequence = ""
-    active_agents = 0
+    agents_cycles_count = 0
     num_cycles = 0
 
     while num_cycles < required_cycles and not is_timeout(start_time, 30):
         for node in nodes:
             start_time_per_agent = time.time()
-            query, sort = elastic_client.build_es_query(term={"agent.name": node.metadata.name})
+            query, sort = elastic_client.build_es_query(
```
[the remark this answers is not in the document]
Comment (gurevichdmitry) [reply tests/commonlib/utils.py:139]: fixed

File: tests/configuration.py [hunk 7]
```diff
@@ -9,24 +9,27 @@
...
-elasticsearch.user = os.getenv('ES_USER', 'elastic')
+elasticsearch.user = os.getenv('ES_USER', 'kube_system')
 elasticsearch.password = os.getenv('ES_PASSWORD', 'changeme')
 elasticsearch.basic_auth = (elasticsearch.user, elasticsearch.password)
 elasticsearch.port = os.getenv('ES_PORT', '9200')
 elasticsearch.protocol = os.getenv('ES_PROTOCOL', 'http')
 elasticsearch.url = f"{elasticsearch.protocol}://{elasticsearch.hosts}:{elasticsearch.port}"
-elasticsearch.cis_index = os.getenv('CIS_INDEX', "*cloud_security_posture.findings*")
+elasticsearch.cis_index = os.getenv(
+    'CIS_INDEX', "*cloud_security_posture.findings*")
```
[the remark this answers is not in the document]
Comment (gurevichdmitry) [reply tests/configuration.py:33]: fixed

[the remark this answers is not in the document]
Comment (gurevichdmitry) [reply tests/configuration.py:33]: fixed

File: tests/integration/tests/conftest.py [hunk 8]
```diff
@@ -1,13 +1,17 @@
 """
 Integration tests setup configurations and fixtures
 """
+# pylint: disable=E0401
```
[the remark this answers is not in the document]
Comment (gurevichdmitry) [reply tests/integration/tests/conftest.py:4]: fixed

File: tests/integration/tests/test_cloudbeat_integration.py [hunk 9]
```diff
@@ -4,6 +4,7 @@
 The following flow is tested:
 Cloudbeat -> ElasticSearch
 """
+# pylint: disable=E0401, E1101
```
[the remark this answers is not in the document]
Comment (gurevichdmitry) [reply tests/integration/tests/test_cloudbeat_integration.py:7]: fixed

File: tests/pylintrc [hunk 10]
```diff
@@ -61,7 +61,7 @@ ignored-modules=
 
 # Python code to execute, usually for sys.path manipulation such as
 # pygtk.require().
-#init-hook=
+init-hook='import sys; sys.path.append(".")'
```
[the remark this answers is not in the document]
Comment (gurevichdmitry) [reply tests/pylintrc:64]: fixed

File: tests/setup.cfg [hunk 11]
```diff
@@ -0,0 +1,6 @@
+[flake8]
+ignore = E123, E203, E302, E501, W503, C901
```
[the remark this answers is not in the document]
Comment (gurevichdmitry) [reply tests/setup.cfg:2]: We can overview flake8 config during pylint baseline task.

File: tests/deploy/cloudbeat-eks-pytest.yaml [hunk 12]
```diff
@@ -0,0 +1,385 @@
+apiVersion: v1
```
[the remark this answers is not in the document]
Comment (gurevichdmitry) [reply tests/deploy/cloudbeat-eks-pytest.yaml:1]: all deployment yamls will be removed during [this task](https://github.com/elastic/security-team/issues/4717)

**Review by oren-zohar — APPROVED** [review]

**Comment by oren-zohar** [comment]
@gurevichdmitry do we need to backport it?

**Comment by kfirpeled** [comment]
> @gurevichdmitry do we need to backport it?

nope, no need to backport tests to branched out versions

**Pull request merged**